src/main/java/zmq/io/net/tcp/TcpConnecter.java

```diff
@@ -267,20 +267,9 @@ private boolean open() throws IOException
             fd = options.selectorChooser.choose(resolved, options).openSocketChannel();
         }
 
-        //  IPv6 address family not supported, try automatic downgrade to IPv4.
-        if (fd == null && resolved.family() == StandardProtocolFamily.INET6 && options.ipv6) {
-            resolved = addr.resolve(false);
-            if (resolved == null) {
-                return false;
-            }
-            // TODO V4 automatic downgrade to IPV4
-            sa = resolved.address();
-            fd = SocketChannel.open();
-        }
-        assert (fd != null);
-
-        //  On some systems, IPv4 mapping in IPv6 sockets is disabled by default.
-        //  Switch it on in such cases.
+        // On some systems, IPv4 mapping in IPv6 sockets is disabled by default.
+        // Switch it on in such cases.
+        // Actually does nothing
```

Comment:
This comment (specifically the "does nothing" part) makes me very curious. It makes me think that maybe the code isn't doing what we're saying it's doing in the comments. Is there some context you could add to the comment?

Reply:
In fact, the method TcpUtils.enableIpv4Mapping is empty. Actually does not have the same meaning in french, I will change that comment

Reply:
But I wonder it that method make any sens at all. Isn't something that was copied directly from C code ? Java already wrap that protocol management. And I think it's too late when this code is reached.

Reply:
Nice -- thanks for the clarification!

I'm lacking the context to know if it makes sense for that method to exist or not... I think in general, it would be fine for us to diverge from the C code in places where Java handles parts of the implementation for us.